pramen/core/src/main/scala/za/co/absa/pramen/core/sql/SqlGeneratorLoader.scala

```diff
@@ -16,50 +16,31 @@
 
 package za.co.absa.pramen.core.sql
 
-import com.typesafe.config.Config
 import org.slf4j.LoggerFactory
+import za.co.absa.pramen.api.sql.{SqlConfig, SqlGenerator}
 
-import java.sql.Connection
-import java.time.LocalDate
-
-trait SqlGenerator {
-  def getDtable(sql: String): String
-
-  def getCountQuery(tableName: String): String
-
-  def getCountQuery(tableName: String, infoDateBegin: LocalDate, infoDateEnd: LocalDate): String
-
-  def getDataQuery(tableName: String, columns: Seq[String], limit: Option[Int]): String
-
-  def getDataQuery(tableName: String, infoDateBegin: LocalDate, infoDateEnd: LocalDate, columns: Seq[String], limit: Option[Int]): String
-
-  def getWhere(dateBegin: LocalDate, dateEnd: LocalDate): String
-
-  /** Returns the date literal for the dialect of the SQL. */
-  def getDateLiteral(date: LocalDate): String
-
-  /**
-    * Quotes an identifier, if needed according to the generator configuration.
-    */
-  def escape(identifier: String): String
+object SqlGeneratorLoader {
+  private val log = LoggerFactory.getLogger(this.getClass)
 
   /**
-    * Always quotes an identifier name with characters specific to SQL dialects.
-    * If the identifier is already wrapped, it won't be double wrapped.
-    * It supports partially quoted identifiers. E.g. '"my_catalog".my table' will be quoted as '"my_catalog"."my table"'.
+    * Loads an SQL generator, If SQL configuration contains a generator class name, it will be loaded.
+    * If not, the generator will be selected based on the driver name based on the internal mapping.
+    * @param driver    The driver class.
+    * @param sqlConfig The SQL configuration.
+    * @return The SQL generator.
     */
-  def quote(identifier: String): String
-
-  def requiresConnection: Boolean = false
-
-  def setConnection(connection: Connection): Unit = {}
-}
+  def getSqlGenerator(driver: String, sqlConfig: SqlConfig): SqlGenerator = {
+    val sqlGenerator = sqlConfig.sqlGeneratorClass match {
+      case Some(clazz) => fromClass(clazz, sqlConfig)
+      case None        => fromDriverName(driver, sqlConfig)
+    }
 
-object SqlGenerator {
-  private val log = LoggerFactory.getLogger(this.getClass)
+    log.info(s"Using SQL generator: ${sqlGenerator.getClass.getCanonicalName}")
+    sqlGenerator
+  }
 
-  def fromDriverName(driver: String, sqlConfig: SqlConfig, extraConfig: Config): SqlGenerator = {
-    val sqlGenerator = driver match {
+  def fromDriverName(driver: String, sqlConfig: SqlConfig): SqlGenerator = {
```

Comment:
Did you leave the methods `fromDriverName` and `fromClass` with public access on purpose?

Reply:
I see, it indeed makes sense to make the private since the object has `getSqlGenerator()` method that can be used in both cases.